sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2258,6 +2258,26 @@ object functions {
     FormatNumber(x.expr, lit(d).expr)
   }
 
+  /**
+   * Convert a string to a number based on the pattern.
+   *
+   * The pattern can consist of the following characters:
+   *   '9': digit position (can be dropped if insignificant)
+   *   '0': digit position (will not be dropped, even if insignificant)
+   *   '.': decimal point (only allowed once)
+   *   ',': group (thousands) separator
+   *   'S': sign anchored to number (uses locale)
+   *   'L': currency symbol (uses locale)
+   *   'D': decimal point (uses locale)
+   *   'G': group separator (uses locale)
+   *
+   * @group string_funcs
+   * @since 3.0.0
+   */
+  def to_number(x: Column, format: String): Column = withExpr {
```

Comment:
Registering the `to_number` function should be enough. Exposing it in Scala API is not necessary.

Reply:
OK. I will remove it.